### ISSUE TITLE
The batchsize of Partialreid experiments 

### ISSUE THREAD

**Issue by zhaoxiaochuang001** [JDAI-CV/fast-reid#534]
When I conduct the  partialeid experiment, I changed the batchsize from Original Settings of partial_market.yml 256 to 64 and run on one gpu. But I got bad results.
Dataset     | Rank-1   | Rank-5   | Rank-10   | mAP   | mINP   |
|:-------------|:---------|:---------|:----------|:------|:-------|
| OccludedREID | 67.90    | 81.00    | 86.00     | 60.23 | 45.12  |

Dataset      | Rank-1   | Rank-5   | Rank-10   | mAP   | mINP   |
|:-------------|:---------|:---------|:----------|:------|:-------|
| PartialiLIDS | 68.91    | 85.71    | 90.76     | 76.50 | 76.50  |

| Dataset     | Rank-1   | Rank-5   | Rank-10   | mAP   | mINP   |
|:------------|:---------|:---------|:----------|:------|:-------|
| PartialREID | 71.00    | 82.33    | 87.33     | 63.95 | 49.24  |

Then I experimented with the original batch_size set to 256 on 4 gpus. The results I got were basically the same as those in your paper.

But I think the change of Batchsize should not have such a big impact on the experimental results. I want to know why?

Here is my experimental setup. I just changed the batchsize to 64 according to the original setting.

CUDNN_BENCHMARK: true
DATALOADER:
  NUM_INSTANCE: 4
  NUM_WORKERS: 8
  SAMPLER_TRAIN: NaiveIdentitySampler
  SET_WEIGHT: []
DATASETS:
  COMBINEALL: false
  NAMES:
  - Market1501
  TESTS:
  - PartialREID
  - PartialiLIDS
  - OccludedREID
INPUT:
  AFFINE:
    ENABLED: false
  AUGMIX:
    ENABLED: false
    PROB: 0.0
  AUTOAUG:
    ENABLED: false
    PROB: 0.0
  CJ:
    BRIGHTNESS: 0.15
    CONTRAST: 0.15
    ENABLED: false
    HUE: 0.1
    PROB: 0.5
    SATURATION: 0.1
  CROP:
    ENABLED: false
    RATIO:
    - 0.75
    - 1.3333333333333333
    SCALE:
    - 0.16
    - 1
    SIZE:
    - 224
    - 224
  FLIP:
    ENABLED: true
    PROB: 0.5
  PADDING:
    ENABLED: false
    MODE: constant
    SIZE: 10
  REA:
    ENABLED: false
    PROB: 0.5
    VALUE:
    - 123.675
    - 116.28
    - 103.53
  RPT:
    ENABLED: false
    PROB: 0.5
  SIZE_TEST:
  - 384
  - 128
  SIZE_TRAIN:
  - 384
  - 128
KD:
  EMA:
    ENABLED: false
    MOMENTUM: 0.999
  MODEL_CONFIG: []
  MODEL_WEIGHTS: []
MODEL:
  BACKBONE:
    ATT_DROP_RATE: 0.0
    DEPTH: 50x
    DROP_PATH_RATIO: 0.1
    DROP_RATIO: 0.0
    FEAT_DIM: 2048
    LAST_STRIDE: 1
    NAME: build_resnet_backbone
    NORM: BN
    PRETRAIN: true
    PRETRAIN_PATH: ''
    SIE_COE: 3.0
    STRIDE_SIZE:
    - 16
    - 16
    WITH_IBN: true
    WITH_NL: false
    WITH_SE: false
  DEVICE: cuda
  FREEZE_LAYERS: []
  HEADS:
    CLS_LAYER: Linear
    EMBEDDING_DIM: 0
    MARGIN: 0.0
    NAME: DSRHead
    NECK_FEAT: before
    NORM: BN
    NUM_CLASSES: 751
    POOL_LAYER: FastGlobalAvgPool
    SCALE: 1
    WITH_BNNECK: true
  LOSSES:
    CE:
      ALPHA: 0.2
      EPSILON: 0.12
      SCALE: 1.0
    CIRCLE:
      GAMMA: 128
      MARGIN: 0.25
      SCALE: 1.0
    COSFACE:
      GAMMA: 128
      MARGIN: 0.25
      SCALE: 1.0
    FL:
      ALPHA: 0.25
      GAMMA: 2
      SCALE: 1.0
    NAME:
    - CrossEntropyLoss
    - TripletLoss
    TRI:
      HARD_MINING: false
      MARGIN: 0.3
      NORM_FEAT: false
      SCALE: 1.0
  META_ARCHITECTURE: PartialBaseline
  PIXEL_MEAN:
  - 123.675
  - 116.28
  - 103.53
  PIXEL_STD:
  - 58.395
  - 57.120000000000005
  - 57.375
  QUEUE_SIZE: 8192
  WEIGHTS: ''
OUTPUT_DIR: projects/PartialReID/logs/test_partial
SOLVER:
  AMP:
    ENABLED: false
  BASE_LR: 0.0007
  BIAS_LR_FACTOR: 1.0
  CHECKPOINT_PERIOD: 20
  CLIP_GRADIENTS:
    CLIP_TYPE: norm
    CLIP_VALUE: 5.0
    ENABLED: false
    NORM_TYPE: 2.0
  DELAY_EPOCHS: 20
  ETA_MIN_LR: 7.0e-07
  FREEZE_ITERS: 0
  GAMMA: 0.1
  HEADS_LR_FACTOR: 1.0
  IMS_PER_BATCH: 64
  MAX_EPOCH: 60
  MOMENTUM: 0.9
  NESTEROV: false
  OPT: Adam
  SCHED: CosineAnnealingLR
  STEPS:
  - 30
  - 55
  WARMUP_FACTOR: 0.1
  WARMUP_ITERS: 500
  WARMUP_METHOD: linear
  WEIGHT_DECAY: 0.0005
  WEIGHT_DECAY_BIAS: 0.0005
  WEIGHT_DECAY_NORM: 0.0
TEST:
  AQE:
    ALPHA: 3.0
    ENABLED: false
    QE_K: 5
    QE_TIME: 1
  DSR:
    ENABLED: true
  EVAL_PERIOD: 10
  FLIP:
    ENABLED: false
  IMS_PER_BATCH: 128
  METRIC: cosine
  PRECISE_BN:
    DATASET: Market1501
    ENABLED: false
    NUM_ITER: 300
  RERANK:
    ENABLED: false
    K1: 20
    K2: 6
    LAMBDA: 0.3
  ROC:
    ENABLED: false

**Comment by L1aoXingyu** [comment]
Batch size has a great impact on performance. The reason is about batchnorm, learning rate, pk sampler, etc. So we literally keep the single GPU batch size unchanged. If you use batch size = 256 with 4 GPUs, then the batch size with 1 GPU is 64, so the performance is near.

**Comment by zhaoxiaochuang001** [comment]
Okay, thanks.
